src/LinuxCaptureService/LinuxCaptureServiceBase.cpp

```diff
@@ -239,8 +239,15 @@ CaptureServiceBase::CaptureInitializationResult LinuxCaptureServiceBase::DoCaptu
       producer_event_processor_.get(), &tracing_handler};
   MemoryInfoHandler memory_info_handler{producer_event_processor_.get()};
 
-  const CaptureOptions& capture_options =
-      start_stop_capture_request_waiter->WaitForStartCaptureRequest();
+  auto capture_options_or_error = start_stop_capture_request_waiter->WaitForStartCaptureRequest();
+  if (capture_options_or_error.has_error()) {
+    ORBIT_ERROR("An error occurred while waiting for start: %s",
+                capture_options_or_error.error().message());
+    client_capture_event_collector->StopAndWait();
+    TerminateCapture();
+    return CaptureInitializationResult::kSuccess;
```

Comment:
It feels wrong to return kSuccess in this case. Is there no better alternative? If not, could we add an additional enum value for this error case?

Reply:
I added the `kFailure` for this case. And both the `LinuxCaptureService::Capture` and `WindowsCaptureService::Capture` will return the grpc status `INTERNAL` when getting `CaptureInitializationResult::kFailure`.

Reply:
I'm wondering if `kAlreadyInProgress` isn't already also a failure. In that case `kFailure` would be too generic. I don't think long names are a problem. If we need more detail, I would even go for `kFailureWhileWaitingForStart` as it's grammatically more correct.